### PR TITLE
build: upgrade to TypeScript 2.8

### PR DESCRIPTION
**Pull request by LucasSloan**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
Builds with TS 2.7
Issue Number: #23092

## What is the new behavior?
Builds with TS 2.8.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by gkalpak** [comment]
Fixed the PR title, but the commit message has to be updated too :grin:

**Comment by LucasSloan** [comment]
Whoops, taken care of.

File: integration/typings_test_ts28/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "angular-integration",
+  "description": "Assert that users with TypeScript 2.6 can type-check an Angular application",
```
Comment (alxhub) [inline integration/typings_test_ts28/package.json]: I'm guessing this should say Typescript 2.8?

**Review by alxhub — APPROVED** [review]

File: integration/typings_test_ts28/package.json (hunk 1 above)
Comment (LucasSloan) [reply integration/typings_test_ts28/package.json]: That sounds right.  I've also gone back and updated the test for 2.7 to say 2.7, not 2.6.

File: integration/typings_test_ts28/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "compilerOptions": {
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "outDir": "../../dist/typings_test_ts27/",
```
Comment (rafaelss95) [inline integration/typings_test_ts28/tsconfig.json]: `28`?

**Review by IgorMinar — APPROVED** [review]
That was actually a pretty smooth update! Thanks Lucas!

Can you please presubmit this with `g3sync presubmit 23782` (see go/angular-g3sync). Thanks

**Comment by IgorMinar** [comment]
Lucas, I'm labeling this with merge, but we still need g3 presubmit before the caretaker will merge this. As soon as that's done, the change will be picked up by the caretaker and merged.

**Comment by LucasSloan** [comment]
Running now.

**Pull request closed without merging**